include/seqan3/io/sam_file/format_sam_base.hpp

```diff
@@ -459,16 +459,34 @@ inline void format_sam_base::read_header(stream_view_type && stream_view,
                                          ref_seqs_type & /*ref_id_to_pos_map*/)
 {
     auto it = std::ranges::begin(stream_view);
+    auto end = std::ranges::end(stream_view);
+    std::string string_buffer{};
 
-    auto parse_tag_value = [&stream_view, &it, this] (auto & value) // helper function to parse the next tag value
+    auto call_read_field = [&it, &string_buffer, this] (auto const & predicate, auto & value)
     {
-        detail::consume(stream_view | views::take_until_or_throw(is_char<':'>)); // skip tag name
-        std::ranges::next(std::ranges::begin(stream_view));                     // skip ':'
-        read_field(stream_view | views::take_until_or_throw(is_char<'\t'> || is_char<'\n'>), value);
-        it = std::ranges::begin(stream_view);                                    // make sure iterator is up2date
+        string_buffer.clear();
+        while (!predicate(*it))
+        {
+            string_buffer.push_back(*it);
+            ++it;
+        }
+        read_field(string_buffer, value);
+    };
+
+    auto take_until_colon_and_consume = [&it] ()
+    {
+        while (!is_char<':'>(*it))
+            ++it;
+        ++it;
+    };
```

Comment:
`views::take_until(is_char<':'>)` doesn't work here?

Reply:
No, see the issue.
We would make a copy of the `stream_view` and hence not update the size of the orignal `stream_view`